### PR TITLE
F/cap refactor

### PR DESCRIPTION
**Pull request by deirf**
This patches are refactoring capability related functionality of syslog-ng and some functions of syslog-ng will be stricted than before:
- open a file source or destination
- open a socket (ip or unix)
- initialize spoof source

### CODE REVIEW

File: lib/gprocess.c [hunk 1]
```diff
@@ -475,11 +328,21 @@ g_process_set_working_dir(const gchar *cwd)
  * capability set. The process will change its capabilities to this value
  * during startup, provided it has enough permissions to do so.
  **/
-void 
+void
 g_process_set_caps(const gchar *caps)
 {
-  if (!process_opts.caps)
-    process_opts.caps = caps;
+  process_opts.caps = caps;
```
Comment (bazsi) [inline lib/gprocess.c]: We should not overwrite user supplied caps with our defaults. I think a that was the reason for the if

File: configure.ac [hunk 2]
```diff
@@ -1393,7 +1393,7 @@ AC_DEFINE_UNQUOTED(ENABLE_MEMTRACE, `enable_value $enable_memtrace`, [Enable mem
 AC_DEFINE_UNQUOTED(ENABLE_SPOOF_SOURCE, `enable_value $enable_spoof_source`, [Enable spoof source support])
 AC_DEFINE_UNQUOTED(ENABLE_IPV6, `enable_value $enable_ipv6`, [Enable IPv6 support])
 AC_DEFINE_UNQUOTED(ENABLE_TCP_WRAPPER, `enable_value $enable_tcp_wrapper`, [Enable TCP wrapper support])
-AC_DEFINE_UNQUOTED(ENABLE_LINUX_CAPS, `enable_value $enable_linux_caps`, [Enable Linux capability management support])
+AC_DEFINE_UNQUOTED(ENABLE_LINUX_CAPS, `enable_value $has_linux_caps`, [Enable Linux capability management support])
 AC_DEFINE_UNQUOTED(ENABLE_ENV_WRAPPER, `enable_value $enable_env_wrapper`, [Enable environment wrapper support])
```
Comment (bazsi) [inline configure.ac]: Why rename to has_ prefix? All the others use an enable prefix

File: lib/privileged-linux.c [hunk 3]
```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2002-2015 BalaBit IT Ltd, Budapest, Hungary
+ *
```
Comment (bazsi) [inline lib/privileged-linux.c]: The copyright owner is simply "BALABIT" without location

File: lib/privileged-linux.c [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+ * COPYING for details.
+ *
+ */
+
+#include "privileged-linux.h"
+#include "gprocess.h"
+#include "messages.h"
+#include "file-perms.h"
+#include "affile/affile-common.h"
+
```
Comment (bazsi) [inline lib/privileged-linux.c]: What are the dependencies here? My perception is that this is a leaf module not depending on anything but the kernel. I don't get these includes.

File: lib/privileged-linux.c [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+#include <glib.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include "gsockaddr.h"
+#include "gsocket.h"
+
+#if ENABLE_SPOOF_SOURCE
+#include <libnet.h>
+#endif
+
```
Comment (bazsi) [inline lib/privileged-linux.c]: libnet dependency? where does it come from?

File: lib/privileged-linux.c [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+  gint rc;
+
+  if (!prctl(PR_GET_DUMPABLE, 0, 0, 0, 0))
+    {
+      rc = prctl(PR_SET_DUMPABLE, 1, 0, 0, 0);
+      if (rc < 0)
+        g_process_message("Cannot set process to be dumpable; error='%s'", g_strerror(errno));
+    }
+}
+
```
Comment (bazsi) [inline lib/privileged-linux.c]: I think this part should either be left in gprocess or separated as a new interface. enabling core dumps really doesn't have a close connection with capabilities.

File: lib/privileged-linux.c [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+  if ((open_opts->open_flags & O_WRONLY) == O_WRONLY ||
+      (open_opts->open_flags & O_RDWR) == O_RDWR)
+    return __get_rw_file_caps(open_opts);
+
+  if ((open_opts->open_flags & O_RDONLY) == O_RDONLY)
+    return __get_ro_file_caps(open_opts);
+
+  return NULL;
+}
```
Comment (bazsi) [inline lib/privileged-linux.c]: these belong to affile and not here.

**Comment by bazsi** [comment]
I think this branch lacks introducing proper abstractions and even though it helps concentrating some of the code paths, it is a net loss on readability.

:-1:

**Comment by lbudai** [comment]
I'll close this...

**Pull request closed without merging**